couchbase/datadog_checks/couchbase/couchbase.py

```diff
@@ -402,3 +410,78 @@ def extract_seconds_value(self, value):
             unit = 'us'
 
         return float(val) / TO_SECONDS[unit]
+
+    def _collect_index_stats_metrics(self):
+        url = urljoin(self._index_stats_url, INDEX_STATS_METRICS_PATH)
+        try:
+            data = self._get_stats(url)
+        except requests.exceptions.RequestException as e:
+            msg = "Error accessing the Index Statistics endpoint: %s: %s" % (url, str(e))
+            self.log.warning(str(e))
+            self.service_check(INDEX_STATS_SERVICE_CHECK_NAME, AgentCheck.CRITICAL, self._tags, msg)
+            if int(self._version.split(".")[0]) >= 7:
```

Comment:
Can we add more error handling here to make sure this doesn't error if the version is `None` or a different format?

Reply:
Sounds good! I added logic to check if the version is none and also some error handling incase the version format retrieval does not start with an integer. It'll log a warning log line and not break (hopefully)